src/main/java/gov/nasa/pds/tools/label/LocationValidator.java

```diff
@@ -161,130 +161,144 @@ public void validate(URL target) {
+            if (directory.isDirectory()) {
+                LOG.debug("Input url is a directory, will indeed crawl for bundle/collection files {}",
+                    url);
+                // Build two lists of files to ignore so the crawler will only process the latest
+                // Bundle and Collection.
+                ArrayList<Target> ignoreBundleList = AggregateManager.buildBundleIgnoreList(url,
+                    this.labelExtension);
+                ignoreList.addAll(ignoreBundleList);
+                Target latestBundle = AggregateManager.getLatestBundle();
+
+                // Only build collection ignore list if latestBundle is not null. The reason is
+                // a bundle
+                // contains collection and if there is no bundle, then there is no collection
+                // information to be gathered.
+                if (latestBundle != null) {
+                    ArrayList<Target> ignoreCollectionList = AggregateManager.buildCollectionIgnoreList(url,
+                        latestBundle.getUrl(), this.labelExtension, true);
+                    ignoreList.addAll(ignoreCollectionList);
+                    LOG.debug("url,ignoreCollectionList {},{}", url, ignoreCollectionList);
+                    LOG.debug("url,ignoreCollectionList.size() {},{}", url, ignoreCollectionList.size());
+                }
+                LOG.debug("url,latestBundle {},{}", url, latestBundle);
+                LOG.debug("url,ignoreBundleList {},{}", url, ignoreBundleList);
+                LOG.debug("url,ignoreBundleList.size() {},{}", url, ignoreBundleList.size());
+            } else {
+                LOG.debug("Input url is a file, will not crawl for bundle/collection files {}", url);
+            }
         }
-      }
 
-      if (!rule.isApplicable(location)) {
-        LOG.error("'{}' validation style is not applicable for location {}", rule.getCaption(),
-            location);
-        return;
-      }
+        if (!rule.isApplicable(location)) {
+            LOG.error("'{}' validation style is not applicable for location {}", rule.getCaption(),
+                location);
+            return;
+        }
+
+        ValidationTask task = new ValidationTask(listener, ruleContext, targetRegistrar);
+        task.setLocation(location);
+        task.setRule(rule);
+        task.setRuleManager(ruleManager);
+        Crawler crawler = CrawlerFactory.newInstance(url);
+        // Set filter so the crawler will ignore other bundle/collection files that are
+        // not latest.
+        crawler.addAllIgnoreItems(ignoreList);
+
+        ruleContext.setCrawler(crawler);
+        ruleContext.setRule(rule);
+
+        LOG.debug("validate:Submitting task to taskManager location {} rule {} ", location,
+            rule.getCaption());
+
+        // 🟢 Progress Tracking Implementation
+        int fileCounter = 0;
+        int trackProgress = ruleContext.getEveryN(); // Get user-defined tracking interval
```

Comment:
@tloubrieu-jpl 

getEveryN() is used to not test every item in a series of like items. Not a random sampling but a sub-sample of items to say that if checked every N of M then have a C confidence that they are all correct. Normally it is set to 1 making it useless for trackProgress. Changing its value for trackProgress alone here will have potentially undesirable effects elsewhere in validate.

Need to either add a new CLI parameter, fix the progress step size, or use some other mechanism to set trackProgress.